middleware/log_request.go

```diff
@@ -33,11 +33,22 @@ func LogRequest(verbose bool) goa.Middleware {
 				"ctrl", goa.ContextController(ctx), "action", goa.ContextAction(ctx))
 			if verbose {
 				if len(r.Header) > 0 {
+					var suppressed map[string]struct{}
```

Comment:
Please move this block outside of the goa handler definition so that the map isn't recreated on every request. Concurrent map access is safe.